src/podman.rs

```diff
@@ -0,0 +1,157 @@
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use futures_util::{StreamExt, TryStreamExt};
+use podman_api::opts::{ContainerCreateOpts, PullOpts, RegistryAuth};
+use std::path::PathBuf;
+use tar::Archive;
+
+use super::image;
+
+pub struct PodmanImage {
+    pub image: String,
+    pub repo: String,
+    pub tag: String,
+    pub runtime: podman_api::Podman,
+}
+
+#[async_trait]
+impl image::Image for PodmanImage {
```

Comment:
nit: the name of the module is `podman` so having it in the type name as well is redundant. I think we could have this be something like

```
impl runtime::Container for Image
```